src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +18,406 @@ internal class CodeStylePage : AbstractOptionPage
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                var curSetting = optionSet.GetOption(option);
+                editorconfig.AppendLine(curSetting.Value.ToString().ToLower() + ":" + curSetting.Notification.ToString().ToLower());
+            }
+        }
+
+        private static void CSharpCodeStyleOptions_GenerateEditorconfig(OptionSet optionSet, Option<CodeStyleOption<string>> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<CodeStyleOption<string>>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                var curSetting = optionSet.GetOption(option);
+                editorconfig.AppendLine(curSetting.Value.ToString().ToLower() + ":" + curSetting.Notification.ToString().ToLower());
+            }
+        }
+
+        private static void CSharpCodeStyleOptions_GenerateEditorconfig(OptionSet optionSet, Option<CodeStyleOption<ExpressionBodyPreference>> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<CodeStyleOption<ExpressionBodyPreference>>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                var curSetting = optionSet.GetOption(option);
+                if (curSetting.Value == ExpressionBodyPreference.Never)
+                {
+                    editorconfig.AppendLine("false" + ":" + curSetting.Notification.ToString().ToLower());
+                }
+                else
+                {
+                    editorconfig.AppendLine("true" + ":" + curSetting.Notification.ToString().ToLower());
+                }
+            }
+        }
+
+        private static void CSharpFormattingOptions_GenerateEditorconfig(OptionSet optionSet, Option<LabelPositionOptions> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<LabelPositionOptions>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+                var curSetting = optionSet.GetOption(option);
+                if (curSetting == LabelPositionOptions.LeftMost)
+                {
+                    editorconfig.AppendLine("flush_left");
+                }
+                else if (curSetting == LabelPositionOptions.OneLess)
+                {
+                    editorconfig.AppendLine("one_less_than_current");
+                }
+                else
+                {
+                    editorconfig.AppendLine("no_change");
```

Comment:
we likely do not want to have fallthrough cases like this.  if we end up adding a new label option we'll miss this.  We should check all expected values, then throw if we get something unexpected.